### ISSUE TITLE
Consider firing 100% scroll depth event sooner

### ISSUE THREAD

**Issue by aschweigert** [Automattic/newspack-plugin#1606]
Looking at the NTG scroll depth events, we seem to fire the 100% event when the very bottom of the page becomes visible in the viewport: https://github.com/Automattic/newspack-plugin/blob/3106f180afb365d6dca55eb76bca2442d64ff51c/includes/class-analytics.php#L297

While this is accurate in a very literal sense, most publishers will use the 100% scroll depth milestone to indicate that a reader has reached the bottom *of an article* (not the bottom of the _page_). This makes it look like fewer readers complete articles 100% than is likely the case. Pulling an example from the new benchmarking dashboards, we see that the 100% milestone has a pretty dramatic dropoff from the 50% milestone:

![scrolldepth](https://user-images.githubusercontent.com/490703/163255554-66dbd122-c628-4e97-8dcf-bcd73a24b467.png)

Could we maybe figure out a way to fire this event sooner, perhaps when the end of the `entry_content` div is reached instead of waiting for the very bottom of the page?

**Comment by aschweigert** [comment]
Noting in that screenshot that this publisher (and indeed, quite a few publishers from what I've been able to tell) still don't have the NTG tags enabled, something I would argue we should probably enable by default (new issue for that question: #1607)

**Comment by yogeshbeniwal** [comment]
Affecting even more to websites with large number of comments.

**Comment by adekbadek** [comment]
Another issue with the current implementation is that the `25%` event may fire to soon – when an article is relatively short, and is using a large featured image, the 25% of the page might be visible on page load, while the beginning of the article is barely visible from "below the fold".

**Comment by adekbadek** [comment]
Definitely, the question is the technical implementation. It's not easily done with `amp-analytics`, but possible.
